src/commands/game_commands/leaderboard.ts

```diff
@@ -154,83 +156,192 @@ export default class LeaderboardCommand implements BaseCommand {
+                        .first();
+
+                    const medalIcon = ["🥇", "🥈", "🥉"][rank] || "";
+                    const displayName = enrolledPlayer ? enrolledPlayer.display_name : `Rank #${(rank) + 1}`;
+                    const rankOrLevelsGained = permanentLb ? `${getRankNameByLevel(player.level)}` : "levels gained";
+                    return {
+                        name: `${medalIcon} ${displayName}`,
+                        value: `${!permanentLb ? "+" : ""}${friendlyFormattedNumber(player.exp)} EXP | ${permanentLb ? "Level" : ""} ${friendlyFormattedNumber(player.level)} ${rankOrLevelsGained}`,
+                    };
+                }));
+
+                let leaderboardType: string;
+                switch (type) {
+                    case LeaderboardType.GLOBAL:
+                        leaderboardType = "Global";
+                        break;
+                    case LeaderboardType.SERVER:
+                        leaderboardType = `${state.client.guilds.get(messageContext.guildID).name}'s`;
+                        break;
+                    case LeaderboardType.GAME:
+                        leaderboardType = "Current Game's";
+                        break;
+                    default:
+                        break;
+                }
+
+                const durationString = !permanentLb ? ` ${duration[0].toUpperCase()}${duration.slice(1)} ` : " ";
+
+                resolve({
+                    title: bold(`${leaderboardType}${durationString}Leaderboard`),
+                    fields,
+                    timestamp: new Date(),
+                    thumbnail: { url: KmqImages.THUMBS_UP },
+                    footer: { text: chooseRandom(leaderboardQuotes) },
+                });
+            }));
         }
 
-        logger.info(`${getDebugLogHeader(message)} | Leaderboard retrieved`);
-        const fields: Array<Eris.EmbedField> = await Promise.all(topPlayers.map(async (player, relativeRank) => {
-            const rank = relativeRank + offset;
-            const enrolledPlayer = await dbContext.kmq("leaderboard_enrollment")
-                .where("player_id", "=", player.player_id)
-                .first();
-
-            const medalIcon = ["🥇", "🥈", "🥉"][rank] || "";
-            const displayName = enrolledPlayer ? enrolledPlayer.display_name : `Rank #${(rank) + 1}`;
-            return {
-                name: `${medalIcon} ${displayName}`,
-                value: `${friendlyFormattedNumber(player.exp)} EXP | Level ${friendlyFormattedNumber(player.level)} (${getRankNameByLevel(player.level)})`,
-            };
-        }));
-
-        let leaderboardType: string;
-        switch (type) {
-            case LeaderboardType.GLOBAL:
-                leaderboardType = "Global";
+        return embedsFns;
+    }
+
+    private static async getPageCount(messageContext: MessageContext, type: LeaderboardType, duration: LeaderboardDuration): Promise<number> {
```

Comment:
why does getPageCount need to query again? in both cases it's been used, you have access to the result set

Reply:
I can remove one use of this function at:
`if (pageOffset + 1 > await LeaderboardCommand.getPageCount(messageContext, type, duration)) {`
Replacing it with the size of the array given by `getLeaderboardEmbeds`.

However, within `getLeaderboardEmbeds`, we don't have access to the result set (the query is awaited piece by piece in each anonymous function generated), so it still needs to be used there.

Reply:
Make `getLeaderboardEmbeds` return the size of `topPlayersQuery` too